tools/flaky-test-reporter/slack_notification.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// slack_notification.go sends notifications to slack channels
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"net/http"
+	"net/url"
+
+	"github.com/knative/test-infra/shared/testgrid"
+)
+
+const (
+	knativeBotName          = "Knative bot"
```

Comment:
Please don't use EngProd in the bot's name. We call the Prow robot "Knative Prow Robot" and it has a robot icon. I suggest you use the icon for consistency, and rename the bot to something like "Knative Testgrid Robot" maybe? And if possible, fill its profile clearly indicating what it is and what it does (refer to Knative Prow Robot for example).

Reply:
It is now recommended by Slack Web API to use application associated bot tokens. So Mark created a Slack app and named it `Knative Engprod`. The only thing we need from the app is a bot token, which allows us modify the bot display name, as `Knative bot` in this case, so changing this line to `knativeBotName = "Knative Testgrid Robot"` would do the job. I will try to figure out the icon part(which is also configurable with `iconEmoji`)

Reply:
Currently out Slack plan only supports up to 10 apps, so we might not want to rename the app to be `Testgrid` only

Reply:
As discussed in person, the current app is replaced with a new app named "Knative Testgrid Robot", and the default icon of this new app is the red green Knative robot